### PR TITLE
Pluggable component: Secret store

### PR DESCRIPTION
**Pull request by pravinpushkar**
# Description

 - Proto definition
 - client implementation
 - unit tests

## Issue reference

Please reference the issue this PR will close: #5635 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: dapr/proto/components/v1/secretstore.proto [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  MetadataRequest metadata = 1;
+}
+
+// Response from initialization.
+message SecretStoreInitResponse {}
+
+// GetSecretRequest is the message to get secret from secret store.
+message GetSecretRequest {
+  // The name of secret key.
+  string key = 2;
```
Comment (philliphoff) [inline dapr/proto/components/v1/secretstore.proto:51]: Silly question, but should field numbering generally start at `1`?

File: pkg/components/secretstores/pluggable.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	_, err := gss.Client.Init(gss.Context, &proto.SecretStoreInitRequest{
+		Metadata: protoMetadata,
+	})
+	if err != nil {
+		return err
+	}
+
+	// TODO Static data could be retrieved in another way, a necessary discussion should start soon.
```
Comment (ItalyPaleAle) [inline pkg/components/secretstores/pluggable.go:49]: I think this is a fine way to do that IMHO.

**Review by ItalyPaleAle — APPROVED** [review]

File: dapr/proto/components/v1/secretstore.proto (hunk 1 above)
Comment (pravinpushkar) [reply dapr/proto/components/v1/secretstore.proto:51]: Yeah, I numbered it with 2 purposefully while trying to match the runtime/proto definition. I think, to make it consistent I will change with 1.

**Comment by pravinpushkar** [comment]
@artursouza @yaron2 @mukundansundar Can we merge this if it is being planned for 1.12 ? In that way I will try to get the sandbox-sdk PR also ready before 14th.

**Review by artursouza — APPROVED** [review]

**Pull request merged**